docs/source/dev/test.rst

```diff
@@ -0,0 +1,107 @@
+will contain a ``CMakeLists.txt`` file and a ``src`` folder, in which the ``.cpp`` file with the test itself will be located.
+
+A simple ``CMakeLists.txt`` file will look like:
+
+.. code-block::
+    set(_TARGET_NAME "devTest")
+
+    append_recursive_files_add_to_src_group("src/" "src/" "cpp" _FILES_SOURCE)
+
+    alpaka_add_executable(
+        ${_TARGET_NAME}
+        ${_FILES_SOURCE})
+    target_link_libraries(
+        ${_TARGET_NAME}
+        PRIVATE common)
+
+    set_target_properties(${_TARGET_NAME} PROPERTIES FOLDER "test/unit")
+
+    add_test(NAME ${_TARGET_NAME} COMMAND ${_TARGET_NAME} ${_ALPAKA_TEST_OPTIONS})
+
+where the first line sets the name of the test that ``CMake`` will build for execution. Furthermore, it is necessary to add
+the folder in which the test is located to the ``CMakeLists.txt`` file under the ``test/unit`` path:
+
+.. code-block::
+    add_subdirectory("dev/")
+
+to make sure that ``CMake`` will build the test written in the ``.cpp`` file under that directory.
+
+After these steps, it all comes to writing the *actual* test. First of all, the necessary header libraries should be
+included the ``.cpp`` file:
+
+.. code-block:: c++
+    #include <alpaka/dev/Traits.hpp>
+    #include <alpaka/test/acc/TestAccs.hpp>
+    #include <catch2/catch.hpp>
+
+The first ``include`` statement targets specifically the header in which the functionality is included; with the second one
+only accelerators corresponding to enabled backends are selected for testing, while the third one is necessary to integrate
+Catch2.
+
+One or more test cases are then introduced with the ``TEMPLATE_LIST_TEST_CASE`` macro. It takes three arguments:
+* a free form test name (must be unique)
+* a tag
+* the accelerator(s) that must run the test case(s) (i.e. ``alpaka::test::TestAccs`` targets all the accelereators selected
+by the ``TestAccs`` header).
+
+Some aliases might be useful:
+
+.. code-block:: c++
+    using Dev = alpaka::Dev<TestType>;
+    using Pltf = alpaka::Pltf<Dev>;
+
+The template parameter for ``alpaka::Dev`` is ``TestType``, which basically target the specific accelerator that corresponds
+to the backend which is executing the test.
+
+Since the function that must be tested has a device object as argument, it is necessary to define one:
+
+.. code-block:: c++
+    Dev const dev(alpaka::getDevByIdx<Pltf>(0u));
+
+Having all the necessary objects well defined, the function can be called and the result stored in an object:
+
+.. code-block:: c++
+    auto const warpExtent = alpaka::getWarpSize(dev);
+
+Finally, the condition under which the test succeed should be defined with the ``REQUIRE`` macro:
+
+.. code-block:: c++
+    REQUIRE(warpExtent > 0);
+
+Such condition depends on what must be actually tested (i.e. the size of the warp can never be equal or less than 0).
+Furthermore, other additional conditions can be set as well:
+
+.. code-block:: c++
+    REQUIRE(std::is_same_v<decltype(warpExtent), int>);
```

Comment:
IMO that should be a `STATIC_REQUIRE`.

Reply:
About this, I think that it might be useful to discuss it on VC with the others as well, since all the tests that have been written have e REQUIRE rather than STATIC_REQUIRE. Maybe it's worth to change it? (STATIC_REQUIRE is actually useful imho)

Reply:
Since you have an example of `REQUIRE` (runtime test) a few lines above, I would really like to see `STATIC_REQUIRE` here as an example for a compile time test.

Reply:
For someone like me that does not have prior experience with Catch2, it would be useful to have a very quick introduction to the most common checks to use (`CHECK`, `REQUIRE`, `STATIC_REQUIRE`, _etc._)